### PR TITLE
Restructure project

### PR DESCRIPTION
**Pull request by akebrissman**
- Restructured the project by moving the app object and the database file to outside the gateway project
- Fixed an error in the postman file
- Modified the startup sequence so the database and its table is created correct

### CODE REVIEW

**Comment by akebrissman** [comment]
I just realized that the instance folder is covered by git ignore so the database configuration files are not checked in. 

To get it working I had to create a instance folder and add a file flask.cfg to it. 
The file shall include these two settings. 
SQLALCHEMY_DATABASE_URI = "sqlite:///test.db"
SQLALCHEMY_TRACK_MODIFICATIONS = False

File: gateway/models/firebase.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from ..db import db
+from gateway import db
```
Comment (mblomdahl) [inline gateway/models/firebase.py:3]: `from .. import db` + PEP8 import grouping!

File: app.py [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 
 # Call the Application Factory function to construct a Flask application instance
 # using the standard configuration defined in /instance/flask.cfg
 app = create_app('flask.cfg')
-
-
-@app.before_first_request
-def create_tables():
+with app.app_context():
     db.create_all()
```
Comment (mblomdahl) [inline app.py:10]: I think this should be done in the `gateway.create_app()` function. Any reason to put it outside the Flask app setup codes?

**Review by mblomdahl — CHANGES_REQUESTED** [review]
When following the instructions in the README, I first get 

```bash
(venv) EMB2:gateway mblomdahl$ ./app.py
bash: ./app.py: Permission denied
```

(Should probably be made executable with a shebang.)

Then, since I didn't pay too much attention to the comment about the `flask.cfg`, I get the following error:

```bash
(venv) EMB2:gateway mblomdahl$ python app.py 
Traceback (most recent call last):
  File "app.py", line 6, in <module>
    app = create_app('flask.cfg')
  File "/Users/mblomdahl/PyCharm/gateway/gateway/__init__.py", line 22, in create_app
    app.config.from_pyfile(config_filename)
  File "/Users/mblomdahl/PyCharm/gateway/venv/lib/python3.7/site-packages/flask/config.py", line 131, in from_pyfile
    with open(filename, mode="rb") as config_file:
FileNotFoundError: [Errno 2] Unable to load configuration file (No such file or directory): '/Users/mblomdahl/PyCharm/gateway/instance/flask.cfg'
```

It's good practice to include sensible defaults, to make it easy for contributors to hit the ground running. :) So... How about
1. [x] updating README to make it very clear (and easy) to copy-paste in the required getting started contents into this `flask.cfg` and/or
2. [x] adding a empty `./instance/` directory with a `.gitkeep` file inside, or
3. [x] checking in the `flask.cfg` and adding a `.gitignore` rule that makes it simple to *not* accidentally commit other files from that folder (configs containing secrets); 
    ```plain
    # .gitignore
    ./instance/*
    !./instance/flask.cfg
    ```

❓❗️ ⚡️

File: app.py (hunk 2 above)
Comment (akebrissman) [reply app.py:10]: I could not get it working when it was inside the create_app function. 
Would be good to understand more about context.

File: gateway/models/firebase.py (hunk 1 above)
Comment (akebrissman) [reply gateway/models/firebase.py:3]: I am not sure I understand what import grouping is.

Comment (akebrissman) [reply gateway/models/firebase.py:3]: Will fix

Comment (akebrissman) [reply gateway/models/firebase.py:3]: Done

**Pull request merged**